src/core/MRMedia.js

```diff
@@ -0,0 +1,242 @@
+        // threejs material. Separating it out for easier updating after it is loaded.
+        // The texture is filled-in in the connected function.
+        this.texture = null;
+
+        // This is used to aid in the formatting for certain object-fit setups
+        // ex: contain, scale-down
+        this.subMediaMesh = null;
+    }
+
+    /**
+     * @function
+     * @description Calculates the width of the img based on the img tag in the shadow root
+     * @returns {number} - the resolved width
+     */
+    get width() {
+        let width = this.objectFitDimensions?.width;
+        return width > 0 ? width : super.width;
+    }
+
+    /**
+     * @function
+     * @description Calculates the height of the img based on the img tag in the shadow root
+     * @returns {number} - the resolved height
+     */
+    get height() {
+        let height = this.objectFitDimensions?.height;
+        return height > 0 ? height : super.height;
+    }
+
+    get mediaWidth() {
+        // to be filled in in children
+    }
+
+    get mediaHeight() {
+        // to be filled in in children
+    }
+
+    generateNewMediaPlaneGeometry() {
+        if (this.object3D.geometry !== undefined) {
+            this.object3D.geometry.dispose();
+        }
+        this.object3D.geometry = mrjsUtils.geometry.UIPlane(this.width, this.height, this.borderRadii, 18);
+    }
+
+    loadMediaTexture() {
+        // filled in by MRMedia children (MRImage,MRVideo,etc)
+    }
+
+    /**
+     * @function
+     * @description Callback function of MREntity - handles setting up this Image and associated 3D geometry style (from css) once it is connected to run as an entity component.
+     */
+    connected() {
+        this.media.setAttribute('src', mrjsUtils.html.resolvePath(this.getAttribute('src')));
+        this.media.setAttribute('style', 'object-fit:inherit; width:inherit');
+
+        this.objectFitDimensions = { height: 0, width: 0 };
+        if (this.getAttribute('src') !== undefined) {
+            this.computeObjectFitDimensions();
+            this.generateNewMediaPlaneGeometry();
+            this.loadMediaTexture();
+        }
+    }
+
+    _mutationCheck() {
+        // to be filled in by children
+    }
+
+    /**
+     * @function
+     * @description Callback function of MREntity - Updates the image's cover,fill,etc based on the mutation request.
+     * @param {object} mutation - the update/change/mutation to be handled.
+     */
+    mutated(mutation) {
+        super.mutated();
+
+        // moving the if 'mutation' handling check to the children, since
+        // mutations are only understood by their actual type. Any mutation
+        // passed through MRMedia directly is undefined since it is not
+        // a direct element for users.
+        //
+        // those functions can do the if (mutation.type ....) handling and
+        // their specific cases and then call back up to here to run the below functionality.
```

Comment:
thoughts?

i dont really like doing it this way but it doesnt work as a pure super call from the children because of how 'mutation' is passed into the function based on entity type -- but also didnt want to reimplement the mutated for every child so this was a decent go-between

Reply:
super is a good solution!

Reply:
went through this again and looks i /can/ actually just remove the children's functions - must have erroring due to some of the ip work i had while i was setting this up - cleaned up and just using the parent version now ~